src/app/vertical-nav/routing/vertical-nav-routing.demo.html

```diff
@@ -0,0 +1,122 @@
+<!--
+  ~ Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<div class="clr-example">
+    <div class="main-container">
+        <header class="header header-6">
+            <div class="branding">
+                <a href="#">
+                    <clr-icon shape="vm-bug"></clr-icon>
+                    <span class="title">Project Clarity</span>
+                </a>
+            </div>
+            <form class="search">
+                <label for="search_input">
+                    <input id="search_input" type="text" placeholder="Search for keywords...">
+                </label>
+            </form>
+            <div class="settings">
+                <a href="javascript://" class="nav-link nav-icon">
+                    <clr-icon shape="cog"></clr-icon>
+                </a>
+            </div>
+        </header>
+        <div class="content-container">
+            <div class="content-area">
+                <div class="controls">
+                    <button class="btn" (click)="toggleNav()">Toggle Nav</button>
+                    <button class="btn" (click)="toggleGroup()">Toggle Group</button>
+                </div>
+                <router-outlet></router-outlet>
+            </div>
+            <clr-vertical-nav
+                [clrVerticalNavCollapsible]="true"
+                [clrVerticalNavCollapsed]="navCollapsed"
+                (clrVerticalNavCollapsedChange)="updateNavCollapsed($event)">
```

Comment:
yes this was for testing just to make sure everything works. i tried to use two way binding in such cases like you suggested in tree view. i tested the long syntax in some places and the short one at other places.